gtd-timer/Extentions/ErrorHandlingMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.AspNetCore.Http;
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+using gtdtimer.Extentions.Exceptions;
+
+namespace gtdtimer.Extentions
+{
+    public class ErrorHandlingMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public ErrorHandlingMiddleware(){}
+
+        public ErrorHandlingMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await next(context);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(context, ex);
+            }
+        }
+
+        public Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            var exceptionCode = GetExceptionCode(exception);
+            var result = JsonConvert.SerializeObject(new
+            {
+                error = exception.Message,
+                code  = exceptionCode
+            });
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = (int)exceptionCode;
+            return context.Response.WriteAsync(result);
+        }
+        public HttpStatusCode GetExceptionCode(Exception exception)
```

Comment:
this should be private

Reply:
But then I will not be able to test this method.

Reply:
You would.. through HandleExceptionAsync. Look into using DefaultHttpContext as fake context for test.
I think something like this should be possible:

var context = new DefaultHttpContext();
context.Response.Body = new MemoryStream()
await errorHandlingMiddleware.Invoke(context)

and then reading context.Response.Body

Reply:
You might also need to mock ErrorHandlingMiddleware...